src/Disks/IO/AsynchronousBoundedReadBuffer.cpp

```diff
@@ -250,9 +264,18 @@ bool AsynchronousBoundedReadBuffer::nextImpl()
     size_t bytes_read = result.size - result.offset;
     if (bytes_read)
     {
+        if (use_page_cache)
+        {
+            page_cache_cell = result.page_cache_cell;
```

Comment:
[Not sure if it makes much sense but] 
It seems if we add `data` (or better named like `data_begin` or `data_without_offset`) parameter to `IAsynchronousReader::Result result`, which will be set by `ThreadPoolRemoteFSReader` to proper value (in case of page cache and without it), then we can unify both cases and get rid of explicit `use_page_cache` check and remove `page_cache_cell` from `IAsynchronousReader::Result result`, and so it will look like:
```
internal_buffer = Buffer(result.data_without_offset, result.data_without_offset + buffer_size);
working_buffer = Buffer(internal_buffer.begin() + result.offset, internal_buffer.begin() + result.size);
```
(here we know that `buffer_size == page_cache_cell->size()` as I understand, so we can substitute)